filebeat/harvester/log.go

```diff
@@ -44,6 +44,29 @@ func (h *Harvester) Harvest() {
 		return
 	}
 
+	go func() {
+		select {
+		case <-h.prospectorDone:
+			close(h.done)
+		case <-h.done:
+			return
+		}
+
+	}()
+
+	if h.config.CloseTimeout > 0 {
+		go func() {
+			select {
+			case <-h.done:
+				return
+			case <-time.After(h.config.CloseTimeout):
+				close(h.done)
```

Comment:
on error, multiline might create a multiline event with current content. I'm worried about storing an offset right in the middle of a multiline-event, potentially splitting one multiline event into 2 events.